### PR TITLE
nano: updated to nano-4.2

### PR DESCRIPTION
**Pull request by 5schatten**
Changelog
- https://pastebin.com/raw/BL2EhqQL

### CODE REVIEW

File: packages/tools/nano/package.mk [hunk 1]
```diff
@@ -3,11 +3,11 @@
...
 
 PKG_NAME="nano"
-PKG_VERSION="4.0"
-PKG_SHA256="1e2fcfea35784624a7d86785768b772d58bb3995d1aec9176a27a113b1e9bac3"
+PKG_VERSION="4.2"
+PKG_SHA256="1143defce62e391b241252ffdb6e5c1ded56cfe26d46ee81b796abe0ccc45df9"
 PKG_LICENSE="GPL"
 PKG_SITE="http://www.nano-editor.org/"
-PKG_URL="http://ftpmirror.gnu.org/${PKG_NAME}/${PKG_NAME}-${PKG_VERSION}.tar.xz"
+PKG_URL="http://ftpmirror.gnu.org/nano/${PKG_NAME}-${PKG_VERSION}.tar.xz"
```
Comment (MilhouseVH) [inline packages/tools/nano/package.mk]: What's the benefit of removing only one instance of PKG_NAME? Personally, I'd leave the PKG_URL as it is.

Comment (5schatten) [reply packages/tools/nano/package.mk]: I changed it back to the state of v3.2 but I could also remove the second `$PKG_NAME` the benefit for me would be to copy & paste the url to the directory containing the nano source file.

Comment (MilhouseVH) [reply packages/tools/nano/package.mk]: Even after replacing all instances of PKG_NAME with `nano` you still can't cut & paste the url directly unless you replace PKG_VERSION too, so how far should we go with this type of change?

Writing a small, generic script to parse packages correctly and dump variables is a better solution in my opinion than trying to adapt code written for a computer into a more human readable form (and potentially reducing maintainability).

Eg. `pkginfo.sh`:
```
#!/bin/bash
. config/options "$1" || exit
for v in PROJECT DEVICE ARCH PKG_NAME PKG_VERSION PKG_SITE PKG_URL PKG_DIR; do
  echo "${v}=\"${!v}\""
done
```
then run `pkginfo.sh nano`:
```
PROJECT="Generic"
DEVICE=""
ARCH="x86_64"
PKG_NAME="nano"
PKG_VERSION="4.0"
PKG_SITE="http://www.nano-editor.org/"
PKG_URL="http://ftpmirror.gnu.org/nano/nano-4.0.tar.xz"
PKG_DIR="/home/neil/projects/pullrequest_repos/LibreELEC.tv/packages/tools/nano"
```
Now you can cut & paste directly.

Specify PROJECT/DEVICE/ARCH etc. for more complicated multi-version packages such as `linux`, and view the correct version, url etc. without having to decode the package in your head. :)

Comment (5schatten) [reply packages/tools/nano/package.mk]: I meant just a simple usecase like you want to copy & paste `http://ftpmirror.gnu.org/nano/` to read the changelog for example. :sweat_smile:

Comment (CvH) [reply packages/tools/nano/package.mk]: `http://ftpmirror.gnu.org/nano/` would be really nice to have as long we have no "automated" bump script like TB uses 
`PKG_WATCH="https://www.nano-editor.org/dist/v(.*)/ nano-([\d\.]+)\.tar\.xz"` & 
 `uscan` you "need" to manually navigate at those folders from time to time

**Pull request merged**